### PR TITLE
Learning assembler normalize progress

### PR DESCRIPTION
**Pull request by MadaraUchiha**
Fixes two bugs:
1. Max speed was actually MinSpeed+MaxSpeed, now it's correctly MaxSpeed
2. Selected quality correlates to % of progress between min and max, rather than the absolute value of the SpeedFactor.

### CODE REVIEW

File: Source/ProjectRimFactory/SAL3/Things/Assemblers/Special/Building_Assembler_Learning.cs [hunk 1]
```diff
@@ -1,105 +1,114 @@
...
+            }
+        }
+
+        public float Progress => Mathf.Clamp01(manager.GetFactorFor(currentBillReport.bill.recipe) / MaxSpeed);
+
+        /// <summary>
+        /// Maps the 0..1 progress to -.1..1.1 scale
+        /// Slightly favors low quality at low progress, and high quality at high progress.
+        /// </summary>
+        public float NormalizedProgress => (Progress * 1.2f) - 0.1f;
```
Comment (MadaraUchiha) [inline Source/ProjectRimFactory/SAL3/Things/Assemblers/Special/Building_Assembler_Learning.cs:34]: I'm not entirely certain about this part. This is meant to make the curve more pronounced at the edges, tendency towards poorer quality at 0 progress, and tendency towards better quality at 1 progress.

**Comment by MadaraUchiha** [comment]
I'm also not sure why the diff is on the entire file, as far as I can tell it's the same line ending, same indentation, etc. :/

**Comment by zymex22** [comment]
@Sn1p3rr3c0n you have to be the wise one here :)
I have No idea what it is.

**Review by Sn1p3rr3c0n — APPROVED** [review]
this fixes two issues with the calculations.
i'm also not entirely certain about the Normalization maybe we need more parameters in the future when we do more balancing.
for now i would leave it in.

**Pull request merged**